sdk/servicebus/service-bus/test/internal/unit/receiverCommon.spec.ts

```diff
@@ -290,6 +293,68 @@ describe("shared receiver code", () => {
 
       assert.equal(numRetryCalls, 2 + 1);
     });
+
+    it("respects retry options", async () => {
+      const errorMessages: string[] = [];
+      const errorCount = 3;
+      let numRetryCalls = 0;
+
+      const fakeRetry = async <T>(): Promise<T> => {
+        ++numRetryCalls;
+
+        if (numRetryCalls < errorCount + 1) {
+          // force retry<> to get called ${errorCount} times (because
+          // we "failed" and threw exceptions and 1 more time where
+          // we succeed.
+          throw new Error(`Attempt ${numRetryCalls}: Force another call of retry<>`);
+        }
+
+        return Promise.resolve({} as T);
+      };
+
+      const retryDelayInMs = 2000;
+      let previousAttemptTime = Date.now();
+      await retryForever(
+        {
+          logPrefix: "logPrefix",
+          logger: logger,
+          onError: (err) => {
+            errorMessages.push(err.message);
+            if (numRetryCalls > 1) {
+              // not the first attempt
+              const currentTime = Date.now();
+              if (currentTime - previousAttemptTime < retryDelayInMs) {
+                errorMessages.push(
+                  `Unexpected, Should've waited at least ${retryDelayInMs} between attempts`
+                );
+              }
+              previousAttemptTime = currentTime;
+            }
+          },
+          retryConfig: {
+            operation: async () => {
+              ++numRetryCalls;
+
+              return 1;
+            },
+            connectionId: "id",
+            operationType: RetryOperationType.connection,
+            retryOptions: {
+              retryDelayInMs,
+            },
+          },
+        },
+        fakeRetry
+      );
+
+      assert.deepEqual(errorMessages, [
+        "Attempt 1: Force another call of retry<>",
+        "Attempt 2: Force another call of retry<>",
+        "Attempt 3: Force another call of retry<>",
+      ]);
```

Comment:
nit:
```suggestion
      assert.deepEqual(errorMessages, Array(errorCount).fill("Attempt 1: Force another call of retry<>"));
```

Reply:
or not because the message has indices in it, please discard this

Reply:
right. the counter increments